src/EnergyPlus/HeatBalanceKivaManager.cc

```diff
@@ -0,0 +1,555 @@
+        Kiva::GroundOutput::OT_CONV
+      };
+
+      if (fnd.hasPerimeterSurface) {
+        outputMap[Kiva::Surface::ST_SLAB_PERIM] = {
+          Kiva::GroundOutput::OT_FLUX,
+          Kiva::GroundOutput::OT_TEMP,
+          Kiva::GroundOutput::OT_CONV
+        };
+      }
+
+      if (fnd.foundationDepth > 0.0) {
+        outputMap[Kiva::Surface::ST_WALL_INT] = {
+          Kiva::GroundOutput::OT_FLUX,
+          Kiva::GroundOutput::OT_TEMP,
+          Kiva::GroundOutput::OT_CONV
+        };
+      }
+
+      // point surface to corresponding ground intance(s)]
+      kivaInstances.push_back(KivaInstanceMap(foundationInstances[inst],
+        outputMap,surfNum,wallSurfaces,surface.Zone));
+
+      // TODO Kiva: Change for walk-out basements (each floor can point to multiple instances)
+      surfaceMap[surfNum] = {inst, Kiva::Surface::ST_SLAB_CORE};
+
+      for (auto& wl : wallSurfaces) {
+        surfaceMap[wl] = {inst, Kiva::Surface::ST_WALL_INT};
+      }
+
+      inst++;
+
+    }
+
+    surfNum++;
+  }
+
+  for (auto& kv : kivaInstances) {
+    auto& grnd = kv.ground;
+
+    // Adjust for concave features using Boundary Layer Adjustment method
+    grnd.calculateBoundaryLayer();
+    grnd.setNewBoundaryGeometry();
+
+    grnd.buildDomain();
+
+    // TODO Kiva: Add wall surfaces to EIO
+    gio::write( DataGlobals::OutputFileInits, "(A)" ) << "! <Kiva Foundation Name>, Horizontal Cells, Vertical Cells, Total Cells, Floor Surface, Wall Surface(s)";
+		gio::write( DataGlobals::OutputFileInits, "(A,',',I5',',I5',',I5',',A)" ) << foundationInputs[DataSurfaces::Surface(kv.floorSurface).OSCPtr].name << grnd.nX << grnd.nZ << grnd.nX*grnd.nZ << DataSurfaces::Surface(kv.floorSurface).Name;
+
+  }
+
+}
+
+void KivaManager::initKivaInstances() {
+
+  // initialize temperatures at the beginning of run environment
+  if ( DataGlobals::BeginEnvrnFlag ) {
+    for (auto& kv : kivaInstances) {
+      auto& grnd = kv.ground;
+
+      // Start with steady-state solution
+      // TODO Kiva: Accelerated approach? Kusuda?
+      grnd.foundation.numericalScheme = Kiva::Foundation::NS_STEADY_STATE;
+      kv.setBoundaryConditions();
+      grnd.calculate(kv.bcs);
+      grnd.foundation.numericalScheme = Kiva::Foundation::NS_ADI;
+    }
+  }
+}
+
+void KivaManager::calcKivaInstances() {
+  // calculate heat transfer through ground
+  for (auto& kv : kivaInstances) {
+    auto& grnd = kv.ground;
+
+    kv.setBoundaryConditions();
+    grnd.calculate(kv.bcs,DataGlobals::MinutesPerTimeStep*60.);
+    grnd.calculateSurfaceAverages();
+    kv.reportKivaSurfaces();
+    kv.plotDomain();
+
+  }
+}
+
+void KivaInstanceMap::plotDomain() {
+
+  #ifdef GROUND_PLOT
+
+  if (DataEnvironment::Month == 1 && DataEnvironment::DayOfMonth == 2 && DataGlobals::HourOfDay == 2 && DataGlobals::TimeStep == 1) {
+    Kiva::SnapshotSettings ss;
+    ss.dir = DataStringGlobals::outDirPathName + "/snapshot";
+    double& l = ground.foundation.reductionLength2;
```

Comment:
Is there anything wrong with this? I'm using "l" as an alias so I don't have to type out "ground.foundation.reductionLength2" everywhere.